corehq/apps/userreports/templates/userreports/edit_data_source.html

```diff
@@ -197,5 +188,52 @@ <h4 class="modal-title">{% trans 'Are you sure you want to delete this data sour
         </div>
       {% endif %}
     {% endif %}
+    <div id="confirm_rebuild" class="modal fade" tabindex="-1" role="dialog">
+      <div class="modal-dialog" role="document">
+        <div class="modal-content">
+          <div class="modal-header">
+            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+              <span aria-hidden="true">&times;</span>
+            </button>
+            <h4 class="modal-title">
+              {% if use_updated_ucr_naming %}
+                {% trans 'Are you sure you want to rebuild this custom web report source?'%}
+              {% else %}
+                {% trans 'Are you sure you want to rebuild this data source?'%}
+              {% endif %}
+            </h4>
+          </div>
+          <div class="modal-body">
+            <p>
+              {% if use_updated_ucr_naming %}
+                {% trans "Rebuilding this report source will erase all the existing data in the report source and start populating it again. While the rebuild is happening, reports based on this data source will show incomplete data." %}
```

Comment:
@AmitPhulera 
I picked this from the original text you wrote.
Wanted to check with you if we want this message to say "reports based on this **report** source will show incomplete data" or its fine the way it is?

Reply:
I think the idea behind adding that line was to tell user what it would actually do. And if you are doing it unknowingly it's probably a bad thing to do.
I think it should be fine the way it is.

Reply:
Hey @AmitPhulera,
Just confirming we are on the same page here.

> I think the idea behind adding that line was to tell user what it would actually do

We are still keeping that line and showing it to the user in a popup instead of when the user hovers over the button, so they would still see it before they go for a rebuild.

My query here was about the term used. I see we wanted to call "data source" as "report source" as a part of the feature flag you added. It seems you missed changing that in the sentence added where you used "data source" instead of "report source". So I wanted to check if you wanted to change that now or it is okay the way it is.

Reply:
Thanks for clarifying. I am not sure if that feature flag is actually used 🤔  I think it should be fine to keep it as it is.